tests/test_generate_listens.py

```diff
@@ -0,0 +1,94 @@
+import os
+import json
+import tempfile
+import unittest
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+import listenbrainz_spark
+from tests import SparkTestCase
+from listenbrainz_spark import utils
+from listenbrainz_spark import hdfs_connection
+
+from pyspark.sql.types import StructField, StructType, ArrayType, StringType, TimestampType, FloatType, IntegerType, BooleanType
+
+HDFS_CLUSTER_URI = 'hdfs://hadoop-master:9000' # the URI to be used with Spark
+TESTDATA_PATH = '/rec/listenbrainz_spark/testdata/test_listens/' # local path to test listens
+LISTENS_PATH = '/test/listens/' # HDFS path to listens
+
+END_DATE = datetime.utcnow().replace(microsecond=0).replace(day=1) # For listens of the current month
+START_DATE = (END_DATE + relativedelta(months=-1)) # For listens of the previous month.
+
+class GenerateTestData(SparkTestCase):
```

Comment:
This class is not a `TestCase` because we're not testing any real code here. If you need listens for a particular test or a group of tests, you should add them to the testcase class for those listens. I think keeping tests pure in the sense that they have no side-effects (keep hdfs in the same state that it was before the tests ran) is a thing to aim for. 

In listenbrainz, for database test cases, we add the specific data in the `setUp` function of the test cases and delete the data afterwards. Does that make sense?

If you think that you'll have a large number of tests which will need this amount of data, then we should add this code to the setUp function of SparkTestCase, although I'm wary of doing that because it would make tests very slow. I think the best way forward here is to call `create_json_files` and `upload_listens_to_hdfs` in `setUpClass` in `CreateDataframeTestCase` in ` listenbrainz_spark/recommendations/tests/test_dataframe.py`.